src/types/global_interfaces.ts

```diff
@@ -76,6 +76,12 @@ export type SqlWhere = Record<string, null | SqlValue | SqlValue[]>;
 
 export type SqlSelect = string[];
 
+export type QueryResult<Base extends object, Select extends keyof Base> = [
+  Select,
+] extends [never]
+  ? Base
+  : Pick<Base, Select>;
+
```

Comment:
This used to be `[Select] extends [never]`, before lint-staged messed it up.. 

This is needed because [reasons](https://github.com/microsoft/TypeScript/issues/31751#issuecomment-498526919)